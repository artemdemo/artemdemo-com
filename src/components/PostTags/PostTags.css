.post-tags-item {
  display: inline-block;
  margin-right: 5px;
}
  .post-tags-item::after {
    content: ',';
  }
  .post-tags-item:last-of-type::after {
    content: '';
  }
