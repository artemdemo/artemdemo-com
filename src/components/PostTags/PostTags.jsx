import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'gatsby';
import styled from 'styled-components';
import { secondaryText } from '../../services/styles';
import { normalizeTag } from '../../services/utils';

const PostTagsSty = styled.div`
    ${secondaryText}
`;

const PostTagsItemSty = styled.div`
    display: inline-block;
    margin-right: 5px;
    
    &::after {
        content: ',';
    }
    &:last-of-type::after {
        content: '';
    }
`;

const PostTags = (props) => {
    const { tags } = props;
    if (tags.length === 0) {
        return null;
    }
    return (
        <PostTagsSty>
            Tags:&nbsp;
            {tags.map(tag => (
                <PostTagsItemSty
                    key={`post-tags-${normalizeTag(tag)}`}
                >
                    <Link to={`/tags/${normalizeTag(tag)}`}>
                        {tag}
                    </Link>
                </PostTagsItemSty>
            ))}
        </PostTagsSty>
    );
};

PostTags.propTypes = {
    tags: PropTypes.arrayOf(PropTypes.string),
};

PostTags.defaultProps = {
    tags: [],
};

export default PostTags;