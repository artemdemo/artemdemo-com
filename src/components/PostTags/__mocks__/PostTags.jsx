import React from 'react';

const PostTags = props => (
    <div data-mock='PostTags'>
        {props.children}
    </div>
);

export default PostTags;