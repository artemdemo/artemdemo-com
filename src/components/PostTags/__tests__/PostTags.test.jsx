import React from 'react';
import renderer from 'react-test-renderer';
import PostTags from '../PostTags';

jest.mock('../../SecondaryText/SecondaryText');

describe('PostTags', () => {
    it('should render without tags', () => {
        const tree = renderer.create(
            <PostTags />
        ).toJSON();

        expect(tree).toMatchSnapshot();
    });

    it('should render with tags', () => {
        const tree = renderer.create(
            <PostTags tags={['one', 'two']} />
        ).toJSON();

        expect(tree).toMatchSnapshot();
    });
});