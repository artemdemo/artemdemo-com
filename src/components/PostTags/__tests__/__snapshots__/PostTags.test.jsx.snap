// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PostTags should render with tags 1`] = `
<div
  data-mock="SecondaryText"
>
  Tags:Â 
  <div
    data-mock="Link"
  >
    {
  "to": "/tags/one",
  "className": "PostTags__PostTagsItemSty-sc-8r4n58-0 gpaIXg"
}
    one
  </div>
  <div
    data-mock="Link"
  >
    {
  "to": "/tags/two",
  "className": "PostTags__PostTagsItemSty-sc-8r4n58-0 gpaIXg"
}
    two
  </div>
</div>
`;

exports[`PostTags should render without tags 1`] = `null`;
